cmd/rekor-server/app/flags.go

```diff
@@ -27,41 +27,43 @@ type LogRangesFlag struct {
 	Ranges sharding.LogRanges
 }
 
-func (l *LogRangesFlag) Set(s string) error {
-	ranges := strings.Split(s, ",")
-	l.Ranges = sharding.LogRanges{}
+// Set logRangeMap from log ranges flag and tlog_id flag
+func (l *LogRangesFlag) Set(logranges string, tlogID int64) error {
+	if tlogID == 0 {
+		return fmt.Errorf("tlog_id cannot be zero when building a range map")
+	}
 
 	var err error
 	inputRanges := []sharding.LogRange{}
 
-	// Only go up to the second to last one, the last one is special cased beow
-	for _, r := range ranges[:len(ranges)-1] {
-		split := strings.SplitN(r, "=", 2)
-		if len(split) != 2 {
-			return fmt.Errorf("invalid range flag, expected two parts separated by an =, got %s", r)
-		}
-		lr := sharding.LogRange{}
-		lr.TreeID, err = strconv.ParseInt(split[0], 10, 64)
-		if err != nil {
-			return err
-		}
-		lr.TreeLength, err = strconv.ParseInt(split[1], 10, 64)
-		if err != nil {
-			return err
-		}
+	if logranges != "" {
+		ranges := strings.Split(logranges, ",")
+		l.Ranges = sharding.LogRanges{}
 
-		inputRanges = append(inputRanges, lr)
-	}
+		// Add the inactive trees specified by the ranges flag into the range map
+		for _, r := range ranges {
+			split := strings.SplitN(r, "=", 2)
+			if len(split) != 2 {
+				return fmt.Errorf("invalid range flag, expected two parts separated by an =, got %s", r)
+			}
+			lr := sharding.LogRange{}
+			lr.TreeID, err = strconv.ParseInt(split[0], 10, 64)
+			if err != nil {
+				return err
+			}
+			lr.TreeLength, err = strconv.ParseInt(split[1], 10, 64)
+			if err != nil {
+				return err
+			}
 
-	// The last entry is special and should not have a terminating range, because this is active.
-	lastRangeStr := ranges[len(ranges)-1]
-	lastTreeID, err := strconv.ParseInt(lastRangeStr, 10, 64)
-	if err != nil {
-		return err
+			inputRanges = append(inputRanges, lr)
+		}
 	}
 
+	// The last entry is special and should not have a terminating range, because this is active.
+	// It is specified by the tlog_id.
 	inputRanges = append(inputRanges, sharding.LogRange{
```

Comment:
Instead of appending the active ID, we can add a `ActiveTreeID int` to the `LogRangesFlag` struct that'll make it easier to access it

Reply:
@priyawadhwa That's a good idea. This also implies we may have to move methods like `func (l *LogRanges) ResolveVirtualIndex(index int)` and `func (l *LogRanges) TotalLength() int64` to be methods on `logRangesFlag` instead of `LogRanges` in order to work across all shards (active + inactive), does that seem reasonable?

Reply:
I've actually got a better way of doing it, I'm turning the `LogRanges` struct into this:
```
type LogRanges struct {
        inactive []LogRange
        active   int64
}
```
That should be the simplest way, stay tuned!